app/src/main/java/com/everest/moviedb/network/RetrofitClient.kt

```diff
@@ -0,0 +1,36 @@
+package com.everest.moviedb.network
+
+import com.everest.moviedb.utils.API_KEY
+import com.everest.moviedb.utils.BASE_URL
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class RetrofitClient {
+
+    var retrofitService: RetrofitService? = null
+
+    fun getClient(): RetrofitService {
+        val okHttpClient = OkHttpClient.Builder()
+            .addInterceptor { chain -> return@addInterceptor addApiKey(chain) }
+            .build()
+
+        if (retrofitService == null) {
+            retrofitService = Retrofit.Builder()
+                .addConverterFactory(GsonConverterFactory.create())
+                .client(okHttpClient)
+                .baseUrl(BASE_URL).build().create(RetrofitService::class.java)
+        }
+        return retrofitService!!
+    }
+
+    private fun addApiKey(chain: Interceptor.Chain): okhttp3.Response {
```

Comment:
In real time app, we will have more interceptors, so better to move it to separate file